ext/random/random.c

```diff
@@ -261,9 +261,11 @@ PHPAPI php_random_status *php_random_status_copy(const php_random_algo *algo, ph
 
 PHPAPI void php_random_status_free(php_random_status *status, const bool persistent)
 {
-	if (status->state) {
-		pefree(status->state, persistent);
+	if (status == NULL) {
+		return;
 	}
```

Comment:
Can ``status`` ever be NULL or should an assertion be added?

Reply:
See https://chat.stackoverflow.com/transcript/message/55792527#55792527 for context. I'm used to `free()`-style functions being a noop if passed `NULL` and consider it a good thing, because is avoids checking for NULL in every place that frees something that might be absent. In this case there might be no such place, but still this should just work in case there is.

Reply:
I've just restructured the method a little, though. Instead of returning early, it checks for `NULL` before dereferencing `status` and just passes a possible-NULL `status` to `pefree()`. This pattern is a little more common for more complex structs.